src/textual/app.py

```diff
@@ -2909,10 +2909,14 @@ async def invoke_ready_callback() -> None:
                             self._driver.write(
                                 Control.move(-cursor_x, -cursor_y + 1).segment.text
                             )
-                            if inline_no_clear and not not self.app._exit_renderables:
```

Comment:
The "not not" is a bug. I think the reasoning was that if there are errors (in the exit renderables) they should be printed rather than the last frame.